File: Data Structures & Algorithms/search-2d-matrix/submission-0.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
            int i = 0;
      int n = matrix.size();
      int m = matrix[0].size();
      int j = (n * m) - 1;
      int counter = 0;
      int x = 0;
      int seen = 0;
      while (i <= n){
        int mid = n - (n + i) / 2;
        if(matrix[mid][0] == target) return true;
        else if(matrix[mid][0] < target) {
          if(seen == mid) break;
          seen = mid;
          i = mid + 1;
        }else{
          n = mid - 1;
        }
      }
      i = 0;
      while (i <= m){
        int mid = m - (m + i) / 2;
        if(matrix[seen][mid] == target) return true;
        else if(matrix[seen][mid] < target) {
          i = mid + 1;
        }else{
          m = mid - 1;
        }
      }
      return false;
  
    }
};

File: Data Structures & Algorithms/find-the-difference/submission-0.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        mp = {}
        for c in s:
            mp[c] = 1 + mp.get(c,0)
        for c in t:
            mp[c] = mp.get(c,0) - 1
        
        for c in mp:
            if mp[c] < 0:
                return c
        return ""